base/ubi9/entrypoint.sh

```diff
@@ -31,6 +31,49 @@ if ! whoami &> /dev/null; then
     echo "${USER_NAME:-user}:x:$(id -u):0:${USER_NAME:-user} user:${HOME}:/bin/bash" >> /etc/passwd
     echo "${USER_NAME:-user}:x:$(id -u):" >> /etc/group
   fi
+
+  # Try to detect if we are running in a user namespace and if so, update /etc/subuid and /etc/subgid files.
+  # The user namespace is created when `UserNamespacesSupport` feature is enabled and `hostUsers` is set to false in Pod spec.
+  # Typical output of `/proc/self/uid_map`:
+  # 1. When NOT running in a user namespace:
+  #         0          0 4294967295
+  # 2. When running in a user namespace:
+  #         0 1481179136      65536
+  #    or
+  #         0       1000          1
+  #         1       1001      64535
+  # For more details see:
+  # - https://man7.org/linux/man-pages/man7/user_namespaces.7.html
+  # - https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/
+  # - https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+  # - https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/
+  if [ -f /proc/self/uid_map ]; then
+    if ! grep -q '^\s*0\s\+0\s\+\([2-9]\|[1-9][0-9]\+\)' /proc/self/uid_map; then
+      echo "Running in a user namespace, user id: $(id -u)"
+      # By default, the valid UIDs/GIDs is the range 0-65535
```

Comment:
IMHO the upstream https://github.com/devfile/developer-images should also work with CRI-O engine / OpenShift, can we support uids such as `1001250000`?

Reply:
We will use a dedicated SCC to enable container run capabilities [1]
So, the userid will always be lower than 65536

[1] https://github.com/cgruver/ocp-4-18-nested-container-tech-preview